dm/syncer/syncer.go

```diff
@@ -1873,7 +1873,8 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		// we calculate startLocation and endLocation(currentLocation) for Query event here
 		// set startLocation empty for other events to avoid misuse
 		startLocation = binlog.Location{}
-		if ev, ok := e.Event.(*replication.QueryEvent); ok {
+		switch ev := e.Event.(type) {
```

Comment:
i think we can delete TODO in L1872 `TODO: support all event` 🎉

Reply:
i think so too, `QueryEvent` and `RowsEvent` are the major event type.